test/ContentValidationListenerTest.php

```diff
@@ -2338,4 +2338,118 @@ public function testEventNameShouldBeResetToOriginalOnCompletionOfListener($even
     {
         $this->assertEquals('route', $event->getName());
     }
+
+    public function testValidatesBothRouteAndBodyParams()
+    {
+        $services = new ServiceManager();
+        $factory  = new InputFilterFactory();
+
+        $services->setService('FooValidator', $factory->createInputFilter([
+            'from-route' => [
+                'name' => 'from-route',
+                'required' => true,
+                'validators' => [
+                    ['name' => 'Digits'],
+                ],
+            ],
+            'from-body' => [
+                'name' => 'from-body',
+                'required' => true,
+                'validators' => [
+                    ['name' => 'Digits'],
+                ],
+            ],
+        ]));
+        $listener = new ContentValidationListener([
+            'Foo' => ['input_filter' => 'FooValidator'],
+            'validate_route_params' => [
+                'api.rest.test' => ['from-route']
```

Comment:
This is confusing to me. What happens when you validate the body parameters? Does the listener now munge the route params in with the body parameters? If so, that feels like a bit of a violation of separation of input; I would not expect the route parameters now to be in with my validated body parameters.